### PR TITLE
Update my location state, draw markers

### PR DESCRIPTION
**Pull request by kaytran2992**
- Update move camera for onMyLocation clicked
- Improve performance 
- Implement draw markers (not cover selected)

### CODE REVIEW

File: lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt [hunk 1]
```diff
@@ -174,6 +177,10 @@ class W3WMapManager(
         mapState.value.cameraState?.moveToPosition(coordinates, animate)
     }
 
+    fun moveToMyLocation(coordinates: W3WCoordinates) {
```
Comment (phpduy) [inline lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt:180]: I don't think we need to add this new method, it's basically the same as method :
```
fun moveToPosition(coordinates: W3WCoordinates, animate: Boolean)
```

**Review by phpduy — CHANGES_REQUESTED** [review]

File: lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt (hunk 1 above)
Comment (kaytran2992) [reply lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt:180]: I need to update the zoom level to focus on the current location. And the level zoom is different for each Map type.

`        updateCameraPosition(
            CameraPosition(
                LatLng(coordinates.lat, coordinates.lng),
                MY_LOCATION_ZOOM,
                cameraState.position.tilt,
                cameraState.position.bearing
            ), true
        )`

Comment (phpduy) [reply lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt:180]: In that case, I suggest adding the zoom level, title,.. as parameters with default value in the method
```
moveToPosition(coordinates: W3WCoordinates, animate: Boolean, zoom: Float = DEFAULT_ZOOM, tilt: Float = 0, bearing: Float = 0)
```

Comment (phpduy) [reply lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt:180]: I still think having this new method has 2 drawbacks:
    - We have two methods with basically the same functionality: Move the camera to a specific Coordinate.
    - The new method `moveToMyLocation` quite forces the camera to move a position with a predefined zoom level, tilt, and bear value without having a way to customize those params.

**Review by phpduy — CHANGES_REQUESTED** [review]

File: lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt (hunk 1 above)
Comment (kaytran2992) [reply lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt:180]: I see your point. Let me update the camera state and remove the CameraPosition data class too.

**Review by phpduy — APPROVED** [review]
Nice one!

**Pull request merged**